### PR TITLE
Feature/edit account

### PR DESCRIPTION
**Pull request by hakobyaann**

### CODE REVIEW

File: Birthday/Birthday/Modules/Profile/Profile/Model/ProfileViewModeling.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//
+//  ProfileViewModeling.swift
+//  Birthday
+//
+//  Created by Anna Hakobyan on 23.10.24.
+//
+
+import Foundation
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Profile/Profile/Model/ProfileViewModeling.swift]: remove this file

File: Birthday/Birthday/SubViews/NavigationBar.swift [hunk 2]
```diff
@@ -38,5 +41,7 @@ struct NavigationBar: View {
 }
 
 #Preview {
-  NavigationBar()
+  NavigationBar {
```
Comment (nellyAslanyan) [inline Birthday/Birthday/SubViews/NavigationBar.swift]: let have this in one line

File: Birthday/Birthday/SubViews/InputField.swift [hunk 3]
```diff
@@ -17,21 +17,24 @@ struct InputField: View {
 
   private var placeholderText: String = ""
   private var isSecureField: Bool = false
+  private var backgroundColor: Color = .snow
```
Comment (nellyAslanyan) [inline Birthday/Birthday/SubViews/InputField.swift]: can be let

File: Birthday/Birthday/Modules/Profile/Profile/Viewmodel/ProfileViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+//
+//  Created by Anna Hakobyan on 23.10.24.
+//
+
+import Foundation
+import Combine
+
+final class ProfileViewModel: ProfileViewModeling {
+  
+  var id: UUID
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Profile/Profile/Viewmodel/ProfileViewModel.swift:13]: where do you need this id?

File: Birthday/Birthday/Modules/Profile/Profile/Model/ProfileViewModeling.swift (hunk 1 above)
Comment (hakobyaann) [reply Birthday/Birthday/Modules/Profile/Profile/Model/ProfileViewModeling.swift]: This file is not empty and it is necessary for creating the Profile screen, here is the content:
protocol ProfileViewModeling: ObservableObject {

  var isLoading: Bool { get set }
  var profileData: ProfileModel { get set }

  func getProfileData()

}
, and here is the usage:
struct ProfileScreen<T: **ProfileViewModeling**>: View {...

File: Birthday/Birthday/SubViews/CircularImage.swift [hunk 5]
```diff
@@ -7,35 +7,95 @@
 
 import SwiftUI
 
-struct CircularImage: View {
+struct CircularImage<Placeholder: View>: View {
   
+  var image: Image?
   let imagePath: String
-  var placeholderImage: Image = Image(systemName: "gift.circle")
+  var placeholderView: Placeholder?
```
Comment (nellyAslanyan) [inline Birthday/Birthday/SubViews/CircularImage.swift]: can be let and private

File: Birthday/Birthday/Modules/Profile/Profile/Viewmodel/ProfileViewModel.swift (hunk 4 above)
Comment (hakobyaann) [reply Birthday/Birthday/Modules/Profile/Profile/Viewmodel/ProfileViewModel.swift:13]: We use arguments in the ProfileScreens enum case, for this reason conforming String is not appropriate.

**Review by Aslanyan99n — DISMISSED** [review]

**Review by nellyAslanyan — APPROVED** [review]

**Pull request merged**